docs/modules/ROOT/pages/account-abstraction.adoc

```diff
@@ -0,0 +1,194 @@
+```
+
+NOTE: Using xref:api:utils.adoc#MessageHashUtils-toEthSignedMessageHash-bytes32-[`MessageHashUtils.toEthSignedMessageHash`] wraps the message with an `\x19Ethereum Signed Message:\n` prefix as most wallets do. However, this is not the most recommended approach since it requires the private key owner to sign an opaque message without contextual information. See xref:account-abstraction.adoc#readable_typed_signatures[readable typed signatures].
+
+==== ERC-7562 Validation Rules
+
+To process a bundle of `UserOperations`, bundlers call xref:api:account.adoc#AccountBase-validateUserOp-struct-PackedUserOperation-bytes32-uint256-[`validateUserOp`] on each operation sender to check whether the operation can be executed. However, the bundler has no guarantee that the state of the blockchain will remain the same after the validation phase. To overcome this problematic, https://eips.ethereum.org/EIPS/eip-7562[ERC-7562] proposes a set of limitations to EVM code so that bundlers (or node operators) are protected from unexpected state changes.
+
+These rules outline the requirements for operations to be processed by the canonical mempool.
+
+TIP: Although any Account that breaks such rules may still be processed by a private bundler, developers should keep in mind the centralization tradeoffs of relying on private infrastructure instead of _permissionless_ execution.
+
+===== A note on upgradeability
+
+xref:upgradeable.adoc[Upgradeable Contracts] might easily violate ERC-7562 storage access rules during the validation phase. For example, when upgradeability is present in a module (i.e. an external validator), the account will need to call the proxy and access the implementation address in storage.
+
+The general recommendation for upgradeable accounts is to use a xref:account-abstraction.adoc#Signer[`Signer`] so that the account accesses its own storage during the validation phase, which is allowed by the ERC-7562 rules and preserves the ability of the account of executing operations permissionlessly.
+
+=== Signers
+
+An Account controlled by a signer is a type of account that depends on a https://en.wikipedia.org/wiki/Digital_signature[digital signature] verification algorithm and is similar to Externally Owned Accounts (EOAs) in that they require a private key signature to authenticate it. These accounts can be used with ERC-4337 infrastructure normally and do not violate xref:account-abstraction.adoc#erc_7562_validation_rules[ERC-7562 Validation Rules] when used to validate signatures externally in their upgradeable version.
+
+Generally speaking, blockchains use the properties of https://en.wikipedia.org/wiki/Digital_signature[digital signatures] to authenticate the sender of each transaction. With this in mind, developers can xref:utilities.adoc#checking_signatures_on_chain[verify signatures on-chain] as a way to authenticate user operations for a given off-chain signer (i.e. a private key).
```

Comment:
What kinds of accounts are not controlled by signers? I found this section confusing.

Reply:
Rephrased. I'm solving since this is still kinda present but written differently.